### DISCUSSION TITLE
Add usability improvements

### DISCUSSION THREAD

**Discussion by Disonantemus** [d99kris/nchat#180] [Ideas]
- **URL previews** (with confirmation):
Because it's a lot better to have some info and picture of a cryptic urls, maybe from videos, news, music, etc.
- **Fuzzy searches** (contacts, emoji):
Because I make mistakes every time and fuzzy is my friend.
- **Visible searches** (contacts, emoji) should show the written pattern:
Because now is hidden and is better to now make mistakes.
- **Help pop-up**:
Better for shortcuts discovery to have a pop-up with all keyboard shortcuts, to not have a browser with the _github_ info when using _nchat_. suggested shortcuts: `F1`, `Ctrl-h`. Makes more sense "here" than at the end of `nchat --help`.
- **Spellcheck that highlights unknown words**:
Now just show suggestions, sometimes is faster to just know which words are unknown to spellchecker to correct  that one letter mistake.
- **`Escape` as alternative to _cancel_**:
I think is a lot more natural (muscle memory). I did try with `nchat -k` and get `\33 KEY_CTRL[`, I did change cancel from `key.conf` but has a big delay, like waiting 4 another key.
- **Better File Picker**
Should not show hidden files by default, also is useful to remember last folder.

---

**Extra: Spellcheck with Hunspell (multi dictionaries):**
If you are like me, that English is not your 1st language, and write a lot mixed language in everything, **hunspell** has support for multi language dictionaries, **ispell** can't and **aspell** should, but I can't make it work with some major hassle. **aspell** also has the better suggestions, but is a no-no if I can't use multi dictionaries.

Example:
- Three dictionaries, two different languages and one specific (medical):
`hunspell -d en_US,de_DE,de_medical`

I did some tests and this works for me, just change this line from `ui.conf` to your languages:
~~~
$ grep spell ~/.nchat/ui.conf
spell_check_command=hunspell -d es,en_US
~~~

**Comment by fulalas** [comment]
I agree `Escape` would be much better to exit/cancel dialogs/actions.

And I also confirm the delay when typing `Escape`.

It's odd that the key `Ctrl` appears when we just press `Escape` (when running nchat -k). I also noticed that pressing `Escape` 2 times quickly fires up `\33` straightway and just after a while comes the other `\33`.

**Comment by d99kris** [comment]
Hi @Disonantemus - thanks for suggesting improvements. Ideally I prefer submitting one feature request per item, but we can leave it as-is, with the slight risk of getting less attention. I think the file picker has gotten a small improvement (shades hidden files) but the other items are probably valid things to look into at some point.

**Comment by d99kris** [comment]
Hi @Disonantemus - the following item has been implemented in 5ed45f9 / v5.1.13:

<div type='discussions-op-text'>

<sup>Originally posted by **Disonantemus** January  9, 2024</sup>

- **Visible searches** (contacts, emoji) should show the written pattern:
Because now is hidden and is better to now make mistakes.
</div>

**Comment by d99kris** [comment]
The following item has been implemented in bdf3421 / v5.1.27:

<div type='discussions-op-text'>

<sup>Originally posted by **Disonantemus** January  9, 2024</sup>
- **Better File Picker**
Should not show hidden files by default, also is useful to remember last folder.
